web/src/information_density.ts

```diff
@@ -171,3 +200,259 @@ export function initialize(): void {
     calculate_timestamp_widths();
     determine_container_query_support();
 }
+
+export const information_density_properties_schema = z.enum([
+    "web_font_size_px",
+    "web_line_height_percent",
+]);
+
+export function enable_or_disable_control_buttons($container: JQuery): void {
+    const info_density_properties = z
+        .array(information_density_properties_schema)
+        .parse(["web_font_size_px", "web_line_height_percent"]);
+    for (const property of info_density_properties) {
+        const $button_group = $container.find(`[data-property='${CSS.escape(property)}']`);
+        const $current_elem = $button_group.find<HTMLInputElement>(".current-value");
+        const current_value = Number.parseInt($current_elem.val()!, 10);
+
+        $button_group
+            .find(".default-button")
+            .prop("disabled", current_value === INFO_DENSITY_VALUES_DICT[property].default);
+        $button_group
+            .find(".increase-button")
+            .prop("disabled", current_value >= INFO_DENSITY_VALUES_DICT[property].maximum);
+        $button_group
+            .find(".decrease-button")
+            .prop("disabled", current_value <= INFO_DENSITY_VALUES_DICT[property].minimum);
+    }
+}
+
+const line_height_supported_values = [122, 131, 140, 149, 158];
```

Comment:
This can be a follow-up, but this should probably be computed from `INFO_DENSITY_VALUES_DICT`

Reply:
Added a TODO comment, and moved this to be next to `INFO_DENSITY_VALUES_DICT`